src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMoKN.cpp

```diff
@@ -0,0 +1,123 @@
+ *
+ * \return A new copy of the process.
+ */
+Func_revPoMoKN* Func_revPoMoKN::clone( void ) const
+{
+    
+    return new Func_revPoMoKN( *this );
+}
+
+
+RevBayesCore::TypedFunction< RevBayesCore::RateGenerator >* Func_revPoMoKN::createFunction( void ) const
+{
+    RevBayesCore::TypedDagNode< long                          >* na = static_cast<const Natural              &>( this->args[0].getVariable()->getRevObject() ).getDagNode();
+    RevBayesCore::TypedDagNode< long                          >* ni = static_cast<const Natural              &>( this->args[1].getVariable()->getRevObject() ).getDagNode();
+    RevBayesCore::TypedDagNode<RevBayesCore::Simplex          >* bf = static_cast<const Simplex              &>( this->args[2].getVariable()->getRevObject() ).getDagNode();
+    RevBayesCore::TypedDagNode<RevBayesCore::RbVector<double> >* ex = static_cast<const ModelVector<RealPos> &>( this->args[3].getVariable()->getRevObject() ).getDagNode();
+    RevBayesCore::TypedDagNode<RevBayesCore::RbVector<double> >* fc = static_cast<const ModelVector<RealPos> &>( this->args[4].getVariable()->getRevObject() ).getDagNode();
+
+    if ( ex->getValue().size() != ( (na->getValue())*(na->getValue())-na->getValue() )/2 )
+    {
+        throw RbException("The number of alleles does not match the number of exchangeabilities given: n_exchangeabilities = (n_alleles*n_alleles-n_alleles)/2");
+    }
+    if ( bf->getValue().size() != na->getValue() )
+    {
+        throw RbException("The number of alleles does not match the number of allele frequencies given: n_allele_freq = n_alleles");
+    }
+    if ( fc->getValue().size() != na->getValue() )
+    {
+        throw RbException("The number of alleles does not match the number of fitness coefficients given: n_fit_coeff = n_alleles");
+    }
+
+    RevBayesCore::revPoMoKNRateMatrixFunction* f = new RevBayesCore::revPoMoKNRateMatrixFunction( na, ni, bf, ex, fc );
+    
+    return f;
+}
+
+
+/* Get argument rules */
+const ArgumentRules& Func_revPoMoKN::getArgumentRules( void ) const
+{
+    
+    static ArgumentRules argumentRules = ArgumentRules();
+    static bool          rules_set = false;
+    
+    if ( !rules_set )
+    {
+        argumentRules.push_back( new ArgumentRule( "K"   , Natural::getClassTypeSpec(), "Number of alleles.", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "N"   , Natural::getClassTypeSpec(), "Number of individuals.", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "pi"  , Simplex::getClassTypeSpec(), "Vector of allele frequencies.", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "rho" , ModelVector<RealPos>::getClassTypeSpec(), "Vector of mutation rates: mu=(mu_a0a1,mu_a1a0,mu_a0a2,...).", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
```

Comment:
is this argument rho or mu ? is it a vector of mutation rates or exchangeabilities ? (l45)

Reply:
updated pomo rate matrices argument rules and commited